### ISSUE TITLE
SSL to CloudFlare broken on Macs

### ISSUE THREAD

**Issue by chief1983** [ngld/old-knossos#111]
Seems to affect at least El Capitan through High Sierra.  Known issue, just documenting it here for posterity.  It seems that the networking library in use for downloading mods does not like to connect to CloudFlare via SSL.  The specific error is:

`SSLError(SSLError(1, '[SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:720)')`

**Comment by chief1983** [comment]
https://github.com/requests/requests/issues/2022
https://github.com/scrapy/scrapy/issues/1764

Both of the above shed some light on the issue.  I confirmed that the _ssl.so seems to be linking against the libssl 0.9.8.something with otool:
```
cliff.gordon:~/fs2_open [master]  $ otool -L /Applications/Knossos.app/Contents/MacOS/_ssl.so 
/Applications/Knossos.app/Contents/MacOS/_ssl.so:
	/usr/lib/libssl.0.9.8.dylib (compatibility version 0.9.8, current version 0.9.8)
	/usr/lib/libcrypto.0.9.8.dylib (compatibility version 0.9.8, current version 0.9.8)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 125.2.0)
```
There may be more solutions than what is specifically mentioned in those, but I imagine that somehow using the built-in ssl library with Python 3 and the default system OpenSSL is not going to ever work with CloudFlare's SSL.

**Comment by ngld** [comment]
This is most likely caused by the outdated OpenSSL version. If I'm correct, this will be solved by `0.7.5` which will ship with a newer version of OpenSSL. I'll leave this issue open until I get verification.

**Comment by ngld** [comment]
IIRC this worked fine in `0.7.8`.
